src/commands/Social/pay.ts

```diff
@@ -24,24 +26,42 @@ export default class extends SkyraCommand {
 
 		if (money <= 0) throw message.language.tget('RESOLVER_POSITIVE_AMOUNT');
 
-		await message.author.settings.sync();
-		const currencyBeforePrompt = message.author.settings.get(UserSettings.Money);
-		if (currencyBeforePrompt < money) throw message.language.tget('COMMAND_PAY_MISSING_MONEY', money, currencyBeforePrompt);
+		const { users } = await DbSet.connect();
+		return users.lock([message.author.id, user.id], async (authorID, targetID) => {
+			const settings = await users.ensure(authorID);
 
-		const accepted = await message.ask(message.language.tget('COMMAND_PAY_PROMPT', user.username, money));
+			const currencyBeforePrompt = settings.money;
+			if (currencyBeforePrompt < money) throw message.language.tget('COMMAND_PAY_MISSING_MONEY', money, currencyBeforePrompt);
 
-		const currencyAfterPrompt = message.author.settings.get(UserSettings.Money);
-		if (currencyAfterPrompt < money) throw message.language.tget('COMMAND_PAY_MISSING_MONEY', money, currencyAfterPrompt);
+			const accepted = await message.ask(message.language.tget('COMMAND_PAY_PROMPT', user.username, money));
 
-		return accepted ? this.acceptPayment(message, user, money) : this.denyPayment(message);
-	}
+			await settings.reload();
+			const currencyAfterPrompt = settings.money;
+			if (currencyAfterPrompt < money) throw message.language.tget('COMMAND_PAY_MISSING_MONEY', money, currencyAfterPrompt);
 
-	private async acceptPayment(message: KlasaMessage, user: KlasaUser, money: number) {
-		await user.settings.sync();
+			if (!accepted) return this.denyPayment(message);
 
-		await message.author.decreaseBalance(money);
-		await user.increaseBalance(money);
```

Comment:
Please implement the logic present in these 2 methods in the new implementation.